File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) 
    {
        unordered_map<int, int> count;
        int m = -1;
        for(int i = 0;i<arr.size();i++)
        {
            count[arr[i]]++;
        }
        for( auto it : arr)
        {
            if (count[it] == it && it > m)
                m = it;
        }
        return (m);
        
    }
};

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.py
class Solution(object):
    def findLucky(self, arr):
        freq = Counter(arr)
        m = -1
        print(max(arr))
        for i in range(1, max(arr)+1):
            if i == freq[i] and freq[i] > m :
                m = freq[i]
        return m
        """
        :type arr: List[int]
        :rtype: int
        """
        